client/my-sites/stats/feedback/index.tsx

```diff
@@ -99,22 +100,17 @@ function FeedbackContent( { clickHandler }: FeedbackPropsInternal ) {
 	);
 }
 
-function FeedbackPanel( { isOpen, clickHandler }: FeedbackPropsInternal ) {
+function FeedbackPanel( { isOpen, animationClassName, clickHandler }: FeedbackPropsInternal ) {
 	const translate = useTranslate();
-	const [ animationClassName, setAnimationClassName ] = useState(
-		FEEDBACK_PANEL_ANIMATION_NAME_ENTRY
-	);
 
```

Comment:
It feels like we could achieve the goal of the PR by adding the following lines here:

```JavaScript
	useEffect( () => {
		if ( isOpen ) {
			setAnimationClassName( FEEDBACK_PANEL_ANIMATION_NAME_ENTRY );
		} else {
			setAnimationClassName( FEEDBACK_PANEL_ANIMATION_NAME_EXIT );
		}
	}, [ isOpen ] );
```

Reply:
I think I got what you are talking about, so I put together some quick code to show how it could be done within the floating panel itself.

https://github.com/Automattic/wp-calypso/compare/try/panel-animation?expand=1

Reply:
I agree with @kangzj here, as the animation would be better controlled in the operating component itself. Leaving the animation class toggling to the parent might be coupled. Ideally, the parent should only control if the element is open.

We can also use `isFloatingPanelOpen` to determine the mounting of the `FeedbackPanel`

```
{ isFloatingPanelOpen && <FeedbackPanel clickHandler={ handleButtonClick } /> }
```

and utilize the [React lifecycle](https://stackoverflow.com/a/54655508/5684745) `componentDidMount/componentWillUnmount` to approach the animation.